server/app/auth/oidc/CiviformOidcLogoutActionBuilder.java

```diff
@@ -111,6 +119,15 @@ public CiviformOidcLogoutActionBuilder setPostLogoutRedirectParam(String param)
   @Override
   public Optional<RedirectionAction> getLogoutAction(
       WebContext context, SessionStore sessionStore, UserProfile currentProfile, String targetUrl) {
+    ProfileUtils profileUtils = new ProfileUtils(sessionStore, profileFactory);
+
+    Optional<CiviFormProfile> civiFormProfile = profileUtils.currentUserProfile(context);
+
+    if (civiFormProfile.isPresent()) {
+      JWT idToken = civiFormProfile.get().getProfileData().getIdToken();
+      extraParams.put("id_token_hint", idToken.serialize());
```

Comment:
**question (blocking):** can we be sure the profile will always have an ID token?

Reply:
We can't - I've added an `isPresent()` check.